File: src/main/java/tech/zolhungaj/amqapi/servercommands/objects/PlayerGameState.java
package tech.zolhungaj.amqapi.servercommands.objects;

import com.squareup.moshi.Json;

import java.util.Optional;

public record PlayerGameState (
        Optional<Integer> gameId,
        boolean isSpectator,
        @Json(name = "private") boolean isPrivateRoom,
        @Json(name = "soloGame") Optional<Boolean> isSoloGame,
        @Json(name = "isRanked") Optional<Boolean> isRankedGame,
        boolean inLobby
){
    public PlayerGameState{
        // check if optional because of mapping issues in moshi with null values in inner objects
        if(isSoloGame == null){
            isSoloGame = Optional.empty();
        }
        if(isRankedGame == null){
            isRankedGame = Optional.empty();
        }
    }
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/objects/PlayerGameState.kt
package tech.zolhungaj.amqapi.servercommands.objects

import com.squareup.moshi.Json

@JvmRecord
data class PlayerGameState(
    val gameId: Int?,
    val isSpectator: Boolean,
    @Json(name = "private") val isPrivateRoom: Boolean,
    @Json(name = "soloGame") val isSoloGame: Boolean?,
    @Json(name = "isRanked") val isRankedGame: Boolean?,
    val inLobby: Boolean
)
